src/modules/cloudflare/turnstile/turnstile.service.ts

```diff
@@ -0,0 +1,66 @@
+import { Inject, Injectable } from '@nestjs/common';
+
+import { TurnstileOptions } from './interfaces/turnstile-options.interface';
+import { ITurnstileVerificationResponse } from './types';
+
+@Injectable()
+export class TurnstileService {
+  constructor(
+    @Inject('TurnstileServiceOptions')
+    private readonly options: TurnstileOptions,
+  ) {}
+
+  async turnstileValidateTokenApi(
+    token: string,
+    secretKey: string,
+  ): Promise<ITurnstileVerificationResponse> {
+    const formData = new FormData();
+    const hostVerificationUrl = this.options.host;
+
+    if (!secretKey || !hostVerificationUrl) {
+      throw new Error(
+        'CAPTCHA environment variables are not configured properly.',
+      );
```

Comment:
InternalServerErrorException(
        `CAPTCHA environment variables are not configured properly. Secret Key: ${secretKey} Host Verification Url:${hostVerificationUrl}`,
      );

Reply:
treatment of error improvemented